src/sentry/static/sentry/app/views/alerts/rules/index.tsx

```diff
@@ -0,0 +1,228 @@
+import {addErrorMessage} from 'app/actionCreators/indicator';
+
+import AlertHeader from '../list/header';
+import {isIssueAlert} from '../utils';
+import {TableLayout} from './styles';
+import RuleListRow from './row';
+
+const DEFAULT_SORT: {asc: boolean; field: 'date_added'} = {
+  asc: false,
+  field: 'date_added',
+};
+const DOCS_URL =
+  'https://docs.sentry.io/workflow/alerts-notifications/alerts/?_ga=2.21848383.580096147.1592364314-1444595810.1582160976';
+
+type Props = RouteComponentProps<{orgId: string}, {}> & {
+  organization: Organization;
+};
+
+type State = {
+  ruleList?: IssueAlertRule[];
+};
+
+class AlertRulesList extends AsyncComponent<Props, State & AsyncComponent['state']> {
+  getEndpoints(): [string, string, any][] {
+    const {params, location} = this.props;
+    const {query} = location;
+
+    return [
+      [
+        'ruleList',
+        `/organizations/${params && params.orgId}/combined-rules/`,
+        {
+          query,
+        },
+      ],
+    ];
+  }
+
+  tryRenderEmpty() {
+    const {ruleList} = this.state;
+    if (ruleList && ruleList.length > 0) {
+      return null;
+    }
+
+    return (
+      <EmptyMessage
+        size="medium"
+        icon={<IconCheckmark isCircled size="48" />}
+        title={t('No alert rules exist for these projects.')}
+        description={tct('Learn more about [link:Alerts]', {
+          link: <ExternalLink href={DOCS_URL} />,
+        })}
+      />
+    );
+  }
+
+  handleDeleteRule = async (projectId: string, rule: IssueAlertRule) => {
+    const {params} = this.props;
+    const {orgId} = params;
+    const alertPath = isIssueAlert(rule) ? 'rules' : 'alert-rules';
+
+    try {
+      await this.api.requestPromise(
+        `/projects/${orgId}/${projectId}/${alertPath}/${rule.id}/`,
+        {
+          method: 'DELETE',
+        }
+      );
+      this.reloadData();
+    } catch (_err) {
+      addErrorMessage(t('Error deleting rule'));
+    }
+  };
+
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderList() {
+    const {loading, ruleList = [], ruleListPageLinks} = this.state;
+    const {orgId} = this.props.params;
+    const {query} = this.props.location;
+
+    const allProjectsFromIncidents = new Set(
+      flatten(ruleList?.map(({projects}) => projects))
+    );
+
+    const sort = {
+      ...DEFAULT_SORT,
+      asc: query.asc === '1',
+      // Currently only supported sorting field is 'date_added'
+    };
+
+    return (
+      <React.Fragment>
+        <Panel>
+          <PanelHeader>
+            <TableLayout>
+              <div>{t('Type')}</div>
+              <div>{t('Alert Name')}</div>
+              <div>{t('Project')}</div>
+              <div>{t('Created By')}</div>
+              <div>
+                <StyledSortLink
+                  to={{
+                    pathname: `/organizations/${orgId}/alerts/rules/`,
+                    query: {
+                      asc: sort.asc ? undefined : '1',
```

Comment:
I think here you will need to keep the original parameters in the query.

Tested this locally with a project filter active, the page load sent a backend query with project=1, but when I sorted it just sent asc=1 and I got rules for all projects.

Reply:
thanks, good catch